app/parse.py

```diff
@@ -12,7 +18,64 @@ class Course:
     name: str
     short_description: str
     type: CourseType
+    modules: int
+    topics: int
+    duration: int
+
+
+class LanguageError(Exception):
+    def __init__(self):
+        super().__init__('Looks like this site violates Part 6 of Article 27 '
+                         'of the Law of Ukraine "On Ensuring the Functioning '
+                         'of the Ukrainian Language as the State Language"')
+
+
+def get_course_details(course_item, course_type: CourseType):
```

Comment:
you can split this function into two: getting information from couse_item, parsing detail page